internal/pkg/agent/application/dispatcher/dispatcher.go

```diff
@@ -221,8 +221,8 @@ func (ad *ActionDispatcher) dispatchCancelActions(ctx context.Context, actions [
 func (ad *ActionDispatcher) gatherQueuedActions(ts time.Time) (queued, expired []fleetapi.Action) {
 	actions := ad.queue.DequeueActions()
 	for _, action := range actions {
-		exp, _ := action.Expiration()
-		if ts.After(exp) {
+		exp, err := action.Expiration()
+		if (err == nil || !errors.Is(err, fleetapi.ErrNoExpiration)) && ts.After(exp) {
```

Comment:
We could have a test that this is properly ignored. I sure it is reading this, but the test would preserve the behavior in cases this changes in the future. 

We have recently had a poor experience because of missing regression testing in the agent action handlers :)